src/pages/LoanFinding/LoanFinding.vue

```diff
@@ -181,16 +193,26 @@ export default {
 			enableQFMobileVersion: false,
 			enableAlmostFundedRow: false,
 			HandOrangeIcon,
+			enableLoanRecommendations: false,
 		};
 	},
 	apollo: {
 		preFetch(config, client) {
 			return client.query({
 				query: experimentAssignmentQuery, variables: { id: THREE_LOANS_RECOMMENDED_ROW_EXP_KEY }
-			}).then(() => {
+			}).then(async () => {
+				const loanRecommendationsExp = await client.query({
+					query: experimentAssignmentQuery,
+					variables: { id: LOAN_RECOMMENDATIONS_EXP_KEY }
+				});
+				const useRecommendations = loanRecommendationsExp?.data?.experiment?.version === 'b';
+
 				const recommendedLoansPromise = client.query({
-					query: flssLoansQueryExtended,
-					variables: prefetchedRecommendedLoansVariables
+					query: useRecommendations ? loanRecommendationsQueryExtended : flssLoansQueryExtended,
+					variables: useRecommendations ? {
+						...prefetchedRecommendationsVariables,
+						userId: config.userId || null
```

Comment:
I don't think we use `config.userId` elsewhere, is this the Kiva user ID? What happens if it's a guest user?

Reply:
Does the new loanRecommendations endpoint not handle non-logged in users? That's typically been handled by the service using the information from the auth header, or lack of information if not logged in...

I also couldn't find anywhere we reference the `userId` on `config` but if you can verify it's there after login that would be cool! Otherwise, you can add a `my > id` query somewhere above this to check for login state, where `id` would be the value you need if present.

If you need to pass `uiv` for non-logged in users, you can extract the `{ cookieStore }` in the `preFetch` params and get it from the cookie...

Reply:
Ignore the config access... I just added a userInfo query into the prefetch function. There's a cache read in the created() function as well which I'm hesitant to remove because I don't actually know how to test user state in dev. Do either of you know how to do that?

Reply:
A lack of user id still appears to return loans from the loansRecommended endpoint.

Reply:
Yes, Rob says null is fine.

Reply:
I got the answer to this question by asking Rob, he says the endpoint does not support visitorId and to use null.

Reply:
Sounds great!